pyanaconda/packaging/yumpayload.py

```diff
@@ -1165,6 +1165,24 @@ def groupDescription(self, groupid):
 
             return (group.ui_name, group.ui_description)
 
+    def groupId(self, group_name):
+        """Translate group name to group ID.
+
+        :param group_name: Valid identifier for group specification.
+        :returns: Group ID or empty string if nothing is found.
+        :raise NoSuchGroup: If group_name doesn't exists.
+        """
+        groups = self._yumGroups
+        if not groups:
+            return ""
```

Comment:
I don't want to see `return None`. I think it is straight way to hard debugging because you never know where this will propagate in future, and None is acting as normal value in many ways.

However you are right that the empty string is not the right solution here. So last man standing, use an exception. I don't like it here either for reasons above but it is smallest evil right now. Therefore I'm going to change this to exception.